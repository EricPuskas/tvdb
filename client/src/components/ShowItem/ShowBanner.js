import React from "react";
import NoBannerFound from "../../assets/img/no_banner.png";
import replaceImage from "../../utils/replaceImage";

const ShowBanner = ({ banner, title }) => {
  return (
    <img
      data-test="banner"
      onError={e => replaceImage(e, NoBannerFound)}
      src={banner ? `https://www.thetvdb.com/banners/${banner}` : NoBannerFound}
      alt={title}
    />
  );
};

export default ShowBanner;
