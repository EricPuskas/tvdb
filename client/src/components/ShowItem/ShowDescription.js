import React from "react";

const ShowDescription = ({ overview, maxChars }) => {
  let description;
  if (overview && overview.length > 0) {
    description = overview.substring(0, maxChars);
    if (overview.length > maxChars) description += "...";
  }
  return (
    <div data-test="description">
      <h3>Overview:</h3>
      <p data-test="overview">{description}</p>
    </div>
  );
};

export default ShowDescription;
