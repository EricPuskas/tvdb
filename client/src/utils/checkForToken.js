import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
