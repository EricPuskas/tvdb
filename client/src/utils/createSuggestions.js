const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
