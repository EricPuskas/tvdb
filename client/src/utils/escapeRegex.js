const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
