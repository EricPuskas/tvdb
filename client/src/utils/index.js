import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
