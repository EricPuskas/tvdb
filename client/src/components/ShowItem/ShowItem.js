import React from "react";
import { Link } from "react-router-dom";

import ShowHeader from "./ShowHeader";
import ShowBanner from "./ShowBanner";
import ShowDescription from "./ShowDescription";

import "./ShowItem.scss";

const ShowItem = ({ show }) => {
  if (show) {
    return (
      <div className="ShowItem" data-test="showitem">
        <Link to={`/shows/${show.id}`}>
          <div className="row">
            <div className="col-lg-3 col-12">
              <ShowHeader
                title={show.seriesName}
                firstAired={show.firstAired}
                network={show.network}
              />
            </div>
            <div className="col-lg-9 col-12">
              <ShowBanner banner={show.banner} title={show.seriesName} />
            </div>
            <div className="col-12 overview">
              <ShowDescription maxChars={350} overview={show.overview} />
            </div>
          </div>
        </Link>
      </div>
    );
  } else {
    return null;
  }
};

export default ShowItem;
