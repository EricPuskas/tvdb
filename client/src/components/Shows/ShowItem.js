import React from "react";
import { Link } from "react-router-dom";

import ShowHeader from "./ShowHeader";
import ShowBanner from "./ShowBanner";
import ShowDescription from "./ShowDescription";

import "./Shows.scss";

const ShowItem = ({ show }) => {
  if (show) {
    return (
      <div className="ShowItem">
        <Link to={`/shows/${show.id}`}>
          <div className="row">
            <div className="col-lg-3 col-12">
              <ShowHeader
                title={show.seriesName}
                firstAired={show.firstAired}
                network={show.network}
              />
            </div>
            <div className="col-lg-9 col-12">
              <ShowBanner banner={show.banner} title={show.seriesName} />
            </div>
            <div className="col-12 overview">
              <ShowDescription maxChars={350} overview={show.overview} />
            </div>
          </div>
        </Link>
      </div>
    );
  } else {
    return null;
  }
};

export default ShowItem;
