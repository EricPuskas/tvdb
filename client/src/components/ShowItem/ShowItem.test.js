import React from "react";
import { shallow, mount } from "enzyme";
import { findByAttribute } from "../../utils";
import { BrowserRouter } from "react-router-dom";
import ShowItem from "./ShowItem";

describe("Show Item Component", () => {
  describe("Gets Props", () => {
    const props = {
      show: {
        id: 12345,
        series: "Arrow",
        network: "The CW",
        banner: "graphical/257655-g9.jpg",
        firstAired: "2017-03-31",
        overview: "Lorem Ipsum Etc etc"
      }
    };
    const component = mount(
      <BrowserRouter>
        <ShowItem {...props} />
      </BrowserRouter>
    );
    const wrapper = findByAttribute(component, "showitem");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });
  });

  describe("Gets No Props", () => {
    const component = shallow(<ShowItem />);
    const wrapper = findByAttribute(component, "showitem");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should not render", () => {
      expect(wrapper.length).toBe(0);
    });
  });
});
