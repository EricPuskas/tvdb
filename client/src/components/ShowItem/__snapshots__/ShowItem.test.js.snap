// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Item Component Gets No Props Should pass snapshot test 1`] = `null`;

exports[`Show Item Component Gets Props Should pass snapshot test 1`] = `"<div class=\\"ShowItem\\" data-test=\\"showitem\\"><a href=\\"/shows/12345\\"><div class=\\"row\\"><div class=\\"col-lg-3 col-12\\"><h2>  </h2><hr><div>First Aired: <strong data-test=\\"firstAired\\"><time datetime=\\"1490907600000\\">Mar 31 (2017)</time></strong></div><div>Network: <strong data-test=\\"network\\">The CW</strong></div><hr></div><div class=\\"col-lg-9 col-12\\"><img data-test=\\"banner\\" src=\\"https://www.thetvdb.com/banners/graphical/257655-g9.jpg\\"></div><div class=\\"col-12 overview\\"><div data-test=\\"description\\"><h3>Overview:</h3><p data-test=\\"overview\\">Lorem Ipsum Etc etc</p></div></div></div></a></div>"`;
